40-agile-wallaby/technical_specification.md

```diff
@@ -0,0 +1,52 @@
+# Prepare for Federated Object Storage (Agile Wallaby)
+**Epic Type:** Implementation Epic
+
+## Scope
+
+This epic aims to enhance existing functionality for all services that interact with S3 compatible object storage.
+Currently, only communication with one configured object storage location is supported, but this should be extended to support multiple locations per service.
+
+### Outline:
+
+The proposed changes will touch the following services: Upload Controller (UCS), Interrogation Room (IRS), Internal File Registry (IFRS), Download Controller (DCS) and the event schema repository.
+
+#### Common Service Changes:
+
+The following configuration changes for object storage are proposed:
+Moving from separate `access_key_id`, `secret_access_key` and `endpoint_url` for one storage configuration, a `s3_object_storages` object will now hold the configuration for all available locations.
+Each location is identified by a label and the nested dict holds both the bucket name that shall be accessed as well as the credentials and endpoint URL for the specified bucket at the given location.
+
+```
+s3_object_storages:
+    DKFZ:
+        permanent_bucket: ghga_permanent
+        credentials:
+            s3_access_key_id: test
+            s3_endpoint_url: http://localstack:4566
```

Comment:
Maybe move the endpoint URL out of credentials.

Reply:
Would probably make more sense to rename credentials instead, so we can reuse the existing S3Config from hexkit

Reply:
Sounds good as well